File: step1_start_project/src/main/java/migration/simple/controller/StatsController.java
package migration.simple.controller;

import migration.simple.responses.StatsResponse;
import migration.simple.service.StatsService;
import migration.simple.types.Stats;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController("stats")
public class StatsController {

    private StatsService statsService;

    public StatsController(StatsService statsService) {
        this.statsService = statsService;
    }

    @GetMapping
    public StatsResponse stats() {
        Stats stats = statsService.getStats();

        return new StatsResponse(true, "user stats", stats);
    }
}

File: step3_migration_to_kotlin/src/main/kotlin/migration/simple/controller/StatsController.kt
package migration.simple.controller

import migration.simple.responses.StatsResponse
import migration.simple.service.StatsService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController("stats")
open class StatsController(private val statsService: StatsService) {

    @GetMapping
    open fun stats(): StatsResponse {
        val stats = statsService.getStats()

        return StatsResponse(true, "user stats", stats)
    }
}
